### DISCUSSION TITLE
How does this compile to native code?

### DISCUSSION THREAD

**Discussion by Maniues** [dotnet/runtime#54553] [Q&A] [answered]
yeah, I know I don't know much, but I have a question:
HOW DOES IT COMPILE TO A NATIVE CODE?
Since AOT is an experiment, it shouldn't be used to compile it, right?
It is written mostly in C #, so how is it compiled? Do you use mono or .NET Framework?
*I know what bootstrapping is.*

**Comment by huoyaoyuan** [comment, marked as answer]
The CLR (CoreCLR or Mono) is written in C++. The C# sources are libraries and **not compiled in native**. They are JITed by CLR at runtime.

**Reply by Maniues** [reply]
Oh, sorry. I didn't see the CoreCLR directory.
My last question: Can I compile libraries with AOT?

**Reply by GeraudFabien** [reply]
https://github.com/dotnet/runtimelab/issues/248

https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/readytorun-overview.md
